src/workerd/api/js-rpc-service.c++

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2017-2023 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#include "js-rpc-service.h"
+
+namespace workerd::api {
+
+GetJsRpcTargetEvent::GetJsRpcTargetEvent()
+    : ExtendableEvent("getJsRpcTarget") {};
+
+kj::Promise<WorkerInterface::CustomEvent::Result> GetJsRpcTargetCustomEventImpl::run(
+    kj::Own<IoContext::IncomingRequest> incomingRequest,
+    kj::Maybe<kj::StringPtr> entrypointName) {
+  incomingRequest->delivered();
+  auto targetPaf = kj::newPromiseAndFulfiller<void>();
```

Comment:
nit: I think you should be able to use structured binding here and avoid moving the promise around!

Reply:
https://github.com/cloudflare/workerd/compare/2a92cb472bbf123daeb3c21f5c8e511603ec131a..7a33e501a03d136d4c42eaec57cf98ebfd887934

Thanks!